### PR TITLE
rosetta

### PR DESCRIPTION
**Pull request by siilats**
the rosetta install is not implemented but it should add the share folder

### CODE REVIEW

**Comment by edigaryev** [comment]
Hello Keith 👋

Thank you for your contribution! Could you please elaborate a bit on your use-case here and provide some concrete example of how this change solves it?

The reason I'm asking this is that "*Rosetta doesn’t support the bootstrapping or installation of Intel Linux distributions on Mac computers with Apple silicon using the Virtualization framework*" ([according to the official documentation](https://developer.apple.com/documentation/virtualization/running_intel_binaries_in_linux_vms_with_rosetta)), so this mechanism seems to be pretty limited to running specific Linux binaries, which Tart doesn't support currently, nor does your PR seem to provide support for.

**Comment by jonnybbb** [comment]
In general having Rosetta support enabled will allow to run x86_64 applications inside the arm64 VM. See e.g [here](https://github.com/diddledani/macOS-Linux-VM-with-Rosetta#setting-up-rosetta). Also UTM supports enabling Rosetta see [here](https://docs.getutm.app/advanced/rosetta/).

**Comment by bootstraponline** [comment]
Rosetta support & Linux clipboard sharing would be great for development purposes. A lot of binaries don't have native arm64 versions for Linux.

**Review by edigaryev — CHANGES_REQUESTED** [review]

**Comment by edigaryev** [comment]
@siilats, to keep matters constructive, what do you think about only introducing the Rosetta-specific changes in this PR:

* the code chunk that calls `installRosetta()` (currently none) and the `installRosetta()` itself
* the code chunk that instantiates `VZLinuxRosettaDirectoryShare()` and adds it to the list of `directorySharingDevices`

This will greatly simplify and thus speed up the review process, and for the rest of the features — we can also introduce them into the codebase, but as separate PRs 🙌

**Comment by edigaryev** [comment]
Closing in favor of https://github.com/cirruslabs/tart/pull/324.

File: Sources/tart/VM.swift [hunk 1]
```diff
@@ -301,10 +379,24 @@ class VM: NSObject, VZVirtualMachineDelegate, ObservableObject {
       let sharingDevice = VZVirtioFileSystemDeviceConfiguration(tag: automountTag)
       sharingDevice.share = VZMultipleDirectoryShare(directories: directories)
 
-      configuration.directorySharingDevices = [sharingDevice]
+        configuration.directorySharingDevices.append(sharingDevice)
```
[the remark this answers is not in the document]
Comment (siilats) [reply Sources/tart/VM.swift:382]: extra sharing device for the rosetta support

**Pull request closed without merging**